lib/services/elasticsearch.js

```diff
@@ -2034,37 +2035,28 @@ class ElasticSearch extends Service {
    *
    * @returns {Promise.<Array>} resolve to an array of documents
    */
-  _getAllDocumentsFromQuery ({ index, body, scroll, from, size }) {
-    const
-      client = this._client,
-      documents = [];
+  async _getAllDocumentsFromQuery (esRequest) {
+    let { body: { hits, _scroll_id } } = await this._client.search(esRequest);
 
-    return new Bluebird((resolve, reject) => {
-      client.search(
-        { index, body, scroll, from, size },
-        function getMoreUntilDone(error, { body: { hits, _scroll_id } }) {
-          if (error) {
-            return reject(error);
-          }
+    if (hits.total.value > this._kuzzle.config.limits.documentsFetchCount) {
+      errorsManager.throw('get_limit_exceeded');
+    }
 
-          for (const hit of hits.hits) {
-            documents.push({
-              _id: hit._id,
-              _source: hit._source
-            });
-          }
+    const documents = hits.hits.map(h => ({ _id: h._id, _source: h._source }));
 
-          if (hits.total.value !== documents.length) {
-            client.scroll({
-              scrollId: _scroll_id,
-              scroll
-            }, getMoreUntilDone);
-          }
-          else {
-            resolve(documents);
-          }
-        });
-    });
+    while (hits.total.value !== documents.length) {
+      ({ body: { hits, _scroll_id } } = await this._client.scroll({
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment of "{ body: { hits, _scroll_id } }" from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AAssignmentWithinCondition)